src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -110,11 +113,33 @@ public double balance() throws IOException{
     public void pay(
         final String keygap, final String user,
         final double amount, final String details
-    ) {
-        throw new UnsupportedOperationException(
-            "Not yet implemented. If you can contribute please, do it at "
-            + "https://github.com/amihaiemil/zold-java-client"
+    ) throws IOException {
+        
+        final HttpPost pay = new HttpPost(
+            URI.create(this.baseUri.toString() + "/do-pay")
         );
+        StringBuilder payload = new StringBuilder();
+        payload.append("{");
```

Comment:
@ammaratef45 I think it would be better to use `javax.json` package to construct json payload:
```java
Json.createObjectBuilder()
  .add("keygap", keygap)
  // add other fields
  .build().toString()
```

Reply:
@g4s8 yes, looks more pretty. thanks.